### PR TITLE
[FEAT] foreign bus refactoring

### PR DESCRIPTION
**Pull request by sindhu-106**
Signed-off-by: sindhuja <sindhuja.gajula@intel.com>

### CODE REVIEW

File: config-files/Test.json [hunk 1]
```diff
@@ -0,0 +1,274 @@
+{
```
Comment (paul-amonson) [inline config-files/Test.json]: What is this new file for?

File: config-files/EventSim.json [hunk 2]
```diff
@@ -1,25 +1,38 @@
...
-            "/v1/stream/cray-dmtf-resource-event" : ["events"]
-        }
-    },
-    "rabbitmq" : {
-        "exchangeName" : "simulator",
-        "uri"          : "amqp://127.0.0.1"
-    }
-}
+  "eventsimConfig" : {
+      "SystemManifest": "/opt/ucs/etc/SystemManifest.json",
```
Comment (paul-amonson) [inline config-files/EventSim.json]: Didn't think we needed this anymore?

File: config-files/EventSim.json [hunk 3]
```diff
@@ -1,25 +1,38 @@
...
+      "HWInventoryPath" : "/opt/ucs/etc",
+      "HWInventoryQueryPath" : "/opt/ucs/etc/HwInvQuery_",
+      "HWInventoryDiscStatUrl" : "/Inventory/DiscoveryStatus",
+      "eventCount": 10,
+      "timeDelayMus": 1,
+      "eventRatioSensorToRas": 1,
+      "randomizerSeed": "234"
+  },
+  "networkConfig" : {
+      "network" : "sse",
```
Comment (paul-amonson) [inline config-files/EventSim.json]: Is this still used?

File: docker-deploy/etc/ProviderMonitoringNetworkForeignBus.json [hunk 4]
```diff
@@ -30,7 +30,77 @@
...
+    "dtmfResourceEvents": {
+      "arguments": {
+        "connectPort": 5678,
+        "connectAddress": "10.54.134.150",
+        "urlPath": "/v1/stream/cray-dmtf-resource-event",
+        "connectTimeout": "30",
+        "requestBuilder": "com.intel.dai.monitoring.SSEStreamRequestBuilder",
+        "requestType": "GET",
+        "requestBuilderSelectors": {
+          "stream_id": "dmtfEvents"
```
Comment (paul-amonson) [inline docker-deploy/etc/ProviderMonitoringNetworkForeignBus.json]: Please don't change this file, it is ready for the MS-40 live demo.

File: dai_core/src/main/java/com/intel/dai/exceptions/SimulatorException.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+// Copyright (C) 2020 Intel Corporation
```
Comment (paul-amonson) [inline dai_core/src/main/java/com/intel/dai/exceptions/SimulatorException.java]: Why is a simulator only exception placed in dai_core?

File: docker-deploy/etc/LocationTranslationMap.json [hunk 6]
```diff
@@ -9,7 +9,8 @@
     "x0c0s26b0n0": "X0-S26",
     "x0c0s24b0n0": "X0-S24",
     "x0c0s21b0n0": "X0-S21",
-    "x0c0s11b0n0": "X0-UAN1"
+    "x0c0s11b0n0": "X0-UAN1",
+    "R0-00-CH00-CN0": "R0-00-CH00-CN0"
```
Comment (paul-amonson) [inline docker-deploy/etc/LocationTranslationMap.json]: This line should not be here for the MS-40 demo.

File: foreign_bus/src/main/java/com/intel/dai/eventsim/ApiReqData.java [hunk 7]
```diff
@@ -0,0 +1,160 @@
+package com.intel.dai.eventsim;
```
Comment (paul-amonson) [inline foreign_bus/src/main/java/com/intel/dai/eventsim/ApiReqData.java]: The path is still under foreign bus?

File: config-files/EventSim.json (hunk 3 above)
Comment (sindhu-106) [reply config-files/EventSim.json]: Yes Paul Still i am using it for network connections.
randomizer seed is used to fetch random index from metadata.

File: docker-deploy/etc/ProviderMonitoringNetworkForeignBus.json (hunk 4 above)
Comment (sindhu-106) [reply docker-deploy/etc/ProviderMonitoringNetworkForeignBus.json]: okay paul

**Comment by sindhu-106** [comment]
Addressed changes requested.

> Please make sure that the changes to eventsim-server are included.

Yes these changes are included

**Pull request merged**